resolwe_bio/processes/chip_seq/macs14.yml

```diff
@@ -72,14 +55,37 @@
       label: Negative peaks (XLS)
       type: basic:file
       required: false
+    - name: species
+      label: Species
+      type: basic:string
+    - name: build
+      label: Build
+      type: basic:string
   run:
     runtime: polyglot
     language: bash
     program: |
       NAME=`basename {{ treatment.bam.file }} .bam`
+      
+      SPECIES={{ treatment.species }}
+      {% set dict = {
+        'Homo sapiens': '2.7e9',
+        'Mus musculus': '1.87e9',
+        'Dictyostelium discoideum': '3.4e7',
+        'Drosophila melanogaster': '1.2e8',
+        'Caenorhabditis elegans': '9e7',
+        'Rattus norvegicus': '2e9',
+        }
+      %}
+      GSIZE={{ dict[treatment.species] }}
+      SPECIES_LIST="{% for species in dict.keys() %}{{species}}, {% endfor %}"
+      SPECIES_LIST=${SPECIES_LIST%??} 
+      if [ -z $GSIZE ]; then
```

Comment:
Use jinja syntax for this. Define `SPECIES_LIST` only if you need it (in case there is no matching species). Something like this:
{% if dict.treatment.species %}
  SPECIES_LIST=...
  re-error "{{treatment.species}} is not a valid hybrid species. Accepted species are: ${SPECIES_LIST}."
{% endif %}

Reply:
I agree with Jure on this, please use the Jinja synatax. In addition to Jure's suggestions, this may be useful too:
```
{% set gsize = species_to_gsize[treatment.species]  %}
```